src/rfkill/rfkill.c

```diff
@@ -99,61 +117,42 @@ static int wait_for_initialized(
                 return 0;
         }
 
-        assert_se(sd_device_get_sysname(device, &sysname) >= 0);
+        assert_se(sd_device_get_sysname(device, &data.sysname) >= 0);
 
         /* Wait until the device is initialized, so that we can get
          * access to the ID_PATH property */
 
-        monitor = udev_monitor_new_from_netlink(NULL, "udev");
-        if (!monitor)
-                return log_error_errno(errno, "Failed to acquire monitor: %m");
-
-        r = udev_monitor_filter_add_match_subsystem_devtype(monitor, "rfkill", NULL);
+        r = sd_device_monitor_new(&monitor);
         if (r < 0)
-                return log_error_errno(r, "Failed to add rfkill udev match to monitor: %m");
+                return log_error_errno(r, "Failed to acquire monitor: %m");
 
-        r = udev_monitor_enable_receiving(monitor);
+        r = sd_device_monitor_filter_add_match_subsystem_devtype(monitor, "rfkill", NULL);
         if (r < 0)
-                return log_error_errno(r, "Failed to enable udev receiving: %m");
+                return log_error_errno(r, "Failed to add rfkill device match to monitor: %m");
 
-        watch_fd = udev_monitor_get_fd(monitor);
-        if (watch_fd < 0)
-                return log_error_errno(watch_fd, "Failed to get watch fd: %m");
+        r = sd_device_monitor_start(monitor, device_monitor_handler, &data, "rfkill-device-monitor");
+        if (r < 0)
+                return log_error_errno(r, "Failed to start device monitor: %m");
 
         /* Check again, maybe things changed */
-        r = sd_device_new_from_subsystem_sysname(&d, "rfkill", sysname);
+        r = sd_device_new_from_subsystem_sysname(&d, "rfkill", data.sysname);
         if (r < 0)
                 return log_full_errno(IN_SET(r, -ENOENT, -ENXIO, -ENODEV) ? LOG_DEBUG : LOG_ERR, r,
-                                      "Failed to open device '%s': %m", sysname);
+                                      "Failed to open device '%s': %m", data.sysname);
 
         if (sd_device_get_is_initialized(d, &initialized) >= 0 && initialized) {
                 *ret = TAKE_PTR(d);
                 return 0;
         }
 
-        for (;;) {
-                _cleanup_(sd_device_unrefp) sd_device *t = NULL;
-                const char *name;
-
-                r = fd_wait_for_event(watch_fd, POLLIN, EXIT_USEC);
-                if (r == -EINTR)
-                        continue;
+        while(!data.device) {
+                r = sd_event_run(sd_device_monitor_get_event(monitor), EXIT_USEC);
```

Comment:
It doesn't really matter, but I think it would be a tiny bit nicer if we'd use sd_event_loop() here, and then make the event handler call sd_event_exit() to exit the loop. The only reason being that this means the exit handlers registered with the event loop will be called (not that there are any in this case, but still...)

Anyway, doesn't matter really.